### DISCUSSION TITLE
Improve Docs

### DISCUSSION THREAD

**Discussion by zllkjc** [web-infra-dev/modern.js#5927] [General]
1. We should classify more content by scenario. Like SSR、AssetPrefix、Compatibility....
2. The document should have a clear classification of APIs, such as what version to start supporting, which APIs are experimental, and which are ready to be deprecated.
3. document should be correctly.
4. if some features are recommended for community implementation,  use cases should be clearly listed

**Comment by zllkjc** [comment]
#5154 #5306

**Comment by zllkjc** [comment]
#3680

**Comment by zllkjc** [comment]
#5175

**Comment by zllkjc** [comment]
#5276

**Comment by zllkjc** [comment]
#3353

**Comment by zllkjc** [comment]
Unit Test use case.

**Comment by chenjiahan** [comment]
- Add more Rsbuild new build config to Modern.js document, and link to Rsbuild document for detailed information.
- Mark the deprecated Rsbuild config and provide some migration tips.

@9aoy cc

**Comment by zllkjc** [comment]
- Remove global Proxy in Modern.js docs
- provide some community use case for proxy ?

**Comment by zllkjc** [comment]
- JSX document add inline script use case
- Do we need ESLint docs ?

**Comment by zllkjc** [comment]
- remove Hooks in Modern.js

**Comment by zllkjc** [comment]
- Do we need Reduck or generator in Modern.js docs

**Comment by zllkjc** [comment]
- intergate SSR/SSG/CSR to rendering ?
- intergate Entry to Routing with convention routes ?
- CSS Module is better in `Style` ?
- Add follow sections into 'Optimization' ? (rsbuild docs is already do)
![image](https://github.com/web-infra-dev/modern.js/assets/68810266/39c2d6a8-eea0-4a32-844d-edd8c91a11e1)
  also `code split`
- Add debug for proxy / mock / inspect command ...
